coalib/bears/Bear.py

```diff
@@ -19,38 +21,75 @@
 from coalib.results.TextPosition import ZeroOffsetError
 from coalib.settings.FunctionMetadata import FunctionMetadata
 from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
 from coalib.settings.ConfigurationGathering import get_config_directory
 
 from .meta import bearclass
 
 
-def _is_debugged(bear):
+def _setting_is_enabled(bear, key):
     """
-    Check whether the bear is in debug mode according to its section-settings.
+    Check setting key is in section.
 
     :param bear: Bear object.
-    :return:     True if ``debug_bears`` is ``True`` or if bear name specified
-                 in ``debug_bears`` setting match with the bear parameter.
+    :param key:  Setting key.
+    :return:     ``True`` if setting value is ``True``. Setting object if
+                 setting key is in section else ``False``.
     """
     if not isinstance(bear, Bear):
-        raise ValueError('Positional argument bear is not an instance of '
-                         'Bear class.')
-    if 'debug_bears' not in bear.section:
+        raise ValueError(
+            'Positional argument bear is not an instance of Bear class.')
+    if key is None:
+        raise ValueError('No setting key passed.')
+
+    if key not in bear.section:
         return False
     try:
-        return bool(bear.section['debug_bears'])
+        return bool(bear.section[key])
     except ValueError:
         pass
-    return bear.name.lower() in (
-            map(str.lower, bear.section['debug_bears']))
+    return bear.section[key]
+
+
+def _is_debugged(bear):
+    """
+    Check whether the bear is in debug mode according to its section-settings.
+
+    :param bear: Bear object.
+    :return:     True if ``debug_bears`` is ``True`` or if bear name specified
+                 in ``debug_bears`` setting match with the bear parameter.
+    """
+    setting = _setting_is_enabled(bear, key='debug_bears')
+    if isinstance(setting, bool):
+        return setting
+    else:
+        return bear.name.lower() in map(str.lower, setting)
+
+
+def _is_profiled(bear):
+    """
+    Check whether the bear is in profile mode according to its section-settings.
+
+    :param bear: Bear object.
+    :return:     current working directory if ``profile`` is ``True``, False
+                 if ``profile`` is ``False`` else return directory path
+                 specified in ``profile``.
+    """
+    setting = _setting_is_enabled(bear, key='profile')
+    if setting is True:
```

Comment:
Stop grouping if together when you have a return.
```py
If not setting:
    return False

If setting is True:
    Return ...

# it must be a Setting
return setting.value
```

Reply:
There was a reason behind that 
```bash
>>> section.append(Setting('profile', 'abc'))
>>> if not setting:
...     print('abc')
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'setting' is not defined
>>> setting = section['profile']
>>> if not setting:
...     print('abc')
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/dist-packages/coala_utils/string_processing/StringConverter.py", line 46, in __bool__
    raise ValueError
ValueError
>>> 
```

Reply:
To prevent it from grouping I will make the logic to
```python
    if setting is True:
        return getcwd()
    if isinstance(setting, Setting):
        return setting.value
    return False
```

Reply:
or something like this 
```python
    setting = _setting_is_enabled(bear, key='profile')
    if isinstance(setting, bool):
        return getcwd() if setting else False
    return setting.value
```
**?**